bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java

```diff
@@ -194,9 +194,9 @@ public String getFieldType() {
     @JsonInclude(JsonInclude.Include.NON_NULL)
     @Nullable
     public Boolean isVisible() {
-        if (getEditMode()) {
-            return true;
-        }
+        // if (getEditMode()) {
```

Comment:
we should enhance the editMode check to account for EDS url also (`md.html `or whatever)

Reply:
How are we hitting the aem environment (I assume it's author) with md.html selector?
can you pls share the complete url?

Reply:
Not in favour of returning a different value based on the request. The client should know whether to honour that value or not.

Reply:
this is not true in case of editor load. if there is a public aPI that we can use to recognise the editor load (be it css), then we shoulld use it, other currently, the mechanism in AEM is based on only the request.

Reply:
If we look at the [getFormDefinition](https://github.com/adobe/aem-core-forms-components/commit/2893b0ec22e47f2462b468ef50555361bad573fb#diff-20ac9a5b4488d998e45e2359a2fb591a0dfd3561d68842b054e653bffbf1599dR367) code, we can see the following [line](https://github.com/adobe/aem-core-forms-components/commit/2893b0ec22e47f2462b468ef50555361bad573fb#diff-20ac9a5b4488d998e45e2359a2fb591a0dfd3561d68842b054e653bffbf1599dR373)

```
mapper.writerWithView(Views.Publish.class).writeValue(writer, this);
```

We already have a publish view for runtime. Within that Publish view there is another factor which deals with authoring/runtime which is implicit and the developer has no control over it. What that means is if I have to get correct Form Definition in author, there is no API right now. How are workflows getting the correct definition, or they do not require it right now.

We have three different JSON right now
1. author view
2. publish view for authoring
3. publish view for publishing

Ideally we should have only author view / publish view.

Reply:
I believe it's unlikely for us to process the JSON from the publish view at runtime because it relies on fix from WCM core component. It's important to mention that we haven't conducted test cases for the runtime with the publish view.

Reply:
One possible solution for java API working: We can fix it having a custom seralizer for this property. We have to use  SerializerProvider#getActiveView while serializing. If the view is Author or Publish, we return the actual property as it is. If there is no view, then the default fallback mechanism based on request applies.